src/cli-options.ts

```diff
@@ -258,12 +264,16 @@ function getArgvOptions(
 /**
  * Try to find the npm user agent being used. If we can't detect it, assume npm.
  */
-function getNpmUserAgent(): 'npm' | 'yarn' | 'pnpm' {
+function getNpmUserAgent(): 'npm' | 'yarnClassic' | 'yarnBerry' | 'pnpm' {
   const userAgent = process.env['npm_config_user_agent'];
   if (userAgent !== undefined) {
     const match = userAgent.match(/^(npm|yarn|pnpm)\//);
     if (match !== null) {
-      return match[1] as 'npm' | 'yarn' | 'pnpm';
+      if (match[1] === 'yarn') {
+        return userAgent.includes('yarn/1') ? 'yarnClassic' : 'yarnBerry';
```

Comment:
```suggestion
        return /^yarn\/[01]\./.test(userAgent) ? 'yarnClassic' : 'yarnBerry';
```

- I'm guessing v0 behaves like v1, not like v2/v3?
- Includes the `.` so that we wouldn't treat a future Yarn v10 as classic.